src/tree/AttachedAccountsTreeItem.ts

```diff
@@ -107,12 +111,24 @@ export class AttachedAccountsTreeItem extends AzureParentTreeItem {
 
         const attachedAccounts: AzureTreeItem[] = await this.getAttachedAccounts();
 
-        return attachedAccounts.length > 0 ? attachedAccounts : [new GenericTreeItem(this, {
-            contextValue: 'cosmosDBAttachDatabaseAccount',
-            label: 'Attach Database Account...',
-            commandId: 'cosmosDB.attachDatabaseAccount',
-            includeInTreeItemPicker: true
-        })];
+        if (attachedAccounts.length > 0) {
+            return attachedAccounts;
+        } else {
+            const genericTreeItemWithoutEmulator = new GenericTreeItem(this, {
+                contextValue: 'cosmosDBAttachDatabaseAccount',
+                label: 'Attach Database Account...',
+                commandId: 'cosmosDB.attachDatabaseAccount',
+                includeInTreeItemPicker: true
+            });
+            const genericTreeItemWithEmulator = new GenericTreeItem(this, {
+                contextValue: 'cosmosDBAttachDatabaseAccount',
```

Comment:
```suggestion
                contextValue: 'cosmosDBAttachEmulator',
```

Reply:
Does this context value have to match with 
`{
                    "command": "cosmosDB.attachEmulator",
                    "when": "view == azureDatabasesExplorer && viewItem == cosmosDBAttachedAccountsWithEmulator"
},` ? I am trying understand the behavior of the context value for GenericTreeItem.

Reply:
So the context value for this generic tree item doesn't matter too much - it should just have a name that matches with the purpose of the tree item.

That rule you mentioned only has to do with the "cosmosDB.attachEmulator" command showing up in a context menu. Which we only want for the `AttachedAccountsTreeItem` that has a context value of "cosmosDBAttachedAccountsWithEmulator ".

Since we define `commandId: 'cosmosDB.attachEmulator'` when this generic tree item is created, clicking it will run that command and we don't have to mess with anything in package.json